### PR TITLE
feat: Update substrate v1.4.0

### PR DESCRIPTION
**Pull request by ukint-vs**
Related changes:
- https://github.com/paritytech/polkadot-sdk/pull/1256
- https://github.com/paritytech/polkadot-sdk/pull/1189
- https://github.com/paritytech/polkadot-sdk/pull/2048
- https://github.com/paritytech/polkadot-sdk/pull/2099
- https://github.com/paritytech/substrate/pull/14405

@gear-tech/dev

### CODE REVIEW

**Comment by breathx** [comment]
Please, rebase, cleanup and mention attached migrations in description

**Review by breathx — CHANGES_REQUESTED** [review]
Requesting changes just for avoiding misclick and merge
The PR requires @ekovalev's deep review

**Review by ekovalev — APPROVED** [review]
Great work. Authorship part looks cleaner than before.

File: runtime/vara/src/lib.rs [hunk 1]
```diff
@@ -615,7 +616,11 @@ parameter_types! {
     pub const BondingDuration: sp_staking::EraIndex = 14;
     // 41 eras during which slashes can be cancelled (slightly less than 7 days)
     pub const SlashDeferDuration: sp_staking::EraIndex = 13;
-    pub const MaxNominatorRewardedPerValidator: u32 = 256;
+    pub const MaxExposurePageSize: u32 = 512;
+    // Note: this is not really correct as Max Nominators is (MaxExposurePageSize * page_count) but
+    // this is an unbounded number. We just set it to a reasonably high value, 1 full page
+    // of nominators.
+    pub const MaxNominators: u32 = 512;
```
Comment (breathx) [inline runtime/vara/src/lib.rs:623]: wouldn't this type of tx exceed our non-mandatory limits?
I think its better to keep 256 for both?

File: node/authorship/src/tests.rs [hunk 2]
```diff
@@ -265,7 +269,7 @@ pub fn init() -> (
 
 pub fn create_proposal<A>(
     mut client: Arc<TestClient>,
-    backend: Arc<TestBackend>,
+    _backend: Arc<TestBackend>,
```
Comment (breathx) [inline node/authorship/src/tests.rs:272]: could be removed now?

**Review by breathx — APPROVED** [review]

**Comment by breathx** [comment]
Well done!

Please, don't merge it atm to make sure smooth delivery. Also needs to resolve master branch conflicts and leave here a comment about CI migrations (I remember it was discussed personally, but better to post it)

**Comment by breathx** [comment]
Actualise branch and resolve opened discussions, please @ukint-vs

**Review by breathx — APPROVED** [review]

File: node/authorship/src/tests.rs (hunk 2 above)
Comment (breathx) [reply node/authorship/src/tests.rs:272]: to be solved by @ukint-vs in upcoming PR

**Pull request merged**